js/tabRestore.js

```diff
@@ -0,0 +1,33 @@
+const MAX_TAB_HISTORY_DEPTH = 20
+var tabHistoryStack = Array()
```

Comment:
Great feature! Just a note: why don't you put the two tabHistory-related variables inside the `tabHistory` object?
Probably it's a matter of personal preference but this way you'll connect tightly the things that belong together and won't "expose" any variables irrelevant for the rest of the browser.